### DISCUSSION TITLE
[Unstable] Guardians are trapped in ruins + ALIEN ITEM FIXES INSIDE

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14661] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

**EDIT: SEE THE FOLLOWUP POST BELOW, I CHANGED A LOT OF ITEMS TO TRY AND FIX THE ISSUE**
https://github.com/FakeFishGames/Barotrauma/discussions/14661#discussioncomment-10719327

The new ruins have pathing issues and break the guardian AI:

* Guardians fail to navigate ruins properly, either due to closed doors or tight walls
* Guardians try to flee to a pod for healing and get stuck

Example of guardian trying to reach the pod and getting stuck in the module Alien_chasm
If a guardian comes from the left or the right to heal, it will get stuck on the door and die

Alien_Chasm
![image](https://github.com/user-attachments/assets/5e652721-fc7a-4945-8b72-32d817c3af43)
![image](https://github.com/user-attachments/assets/dffe24d5-d0e4-4fac-b160-210d5f7ecf8c)

Other example: Alien_Chambers4way
Guardians coming from the left can get stuck on the door

![image](https://github.com/user-attachments/assets/abd72a50-77ea-483b-8a73-7c280b72cc3d)

Door blocking the path to/from top
![image](https://github.com/user-attachments/assets/0b679232-062d-4e87-979a-2c161290c832)

Then there are the modules that physically block guardians.
Here's a map of guardian pathing with potential obstructions (doors and walls)
Guardians can only patrol a fraction of the ruins at a time

![image](https://github.com/user-attachments/assets/187a268f-bad8-4d20-b855-89a15253bb80)

The biggest offenders:

Cuts all 4 ways
![image](https://github.com/user-attachments/assets/b19fed3f-c206-48e9-bcba-cbdf2097fcc6)

Cuts top to bottom
![image](https://github.com/user-attachments/assets/51c896fb-f2d5-4a02-8940-a7f893b3222f)

Cuts left to right
![image](https://github.com/user-attachments/assets/98213a12-cf13-4240-9930-fda4d8361d2c)

![image](https://github.com/user-attachments/assets/0b84d225-f5a0-467d-92f6-94e8f264feab)

------------------

There are 2 solutions to this, one easy the other not:

1. Remove those problematic doors and walls on the main path. Most of those are just simple terminal or button checks.

2. Allow aliens to open some doors but not the player:
* Give button properties to unbuttoned doors
* ID lock those doors to use id_guardian
* Give id_guardian tag to guardian weapons so they can open them
* Redo the circuit so it can handle terminals as well as self closing

I could mount a small assembly for solution number 2 if necessary

### Version

v1.6.4.0 (unstable)

**Comment by Ek-Videogames** [comment]
Ruins also suffer from alot of general pathing node issues, either due to the placed paths themselves or the "auto generated" connections between modules.

Simply order a bot to follow you and swim through even the relatively basic hallways and they'll often get stuck or turn around at certain doorways.

**Comment by ubertpendragon** [comment]
# I changed Alienitems.xml to try and include a bunch of important fixes to improve ruins _(and some alien items)_

THE GOAL IS TO ALLOW GUARDIANS TO OPEN DOORS THAT ARE CLOSED FOR THE PLAYER (USING TERMINAL PUZZLES AND WHATNOT), AS WELL AS MAKING CIRCUITS EASIER WHEN EDITING RUINS.

[Fixed alien items.zip](https://github.com/user-attachments/files/17090589/Fixed.alien.items.zip)
Also contains item assemblies to replace existing doors. Make sure to refresh prefabs just in case.


CHANGES:
-Made connectionpanel require a [Locked] item, but can be opened properly in sub editor by default
-Added "guardianitem" tag to guardian weapons
-Added a "button" do non-buttoned doors so guardians can open them (using "guardianitem")
-Added a shutdown pin to alien doors so you can destroy them with terminals
-Added activate_out pin to alien doors (makes circuits easier)
-Added particles and sound when breaking a door
-Disabled the self-closing statuseffects on doors (this messed with circuits)
-Fixed many sprites on doors being outright missing, having the wrong depth or not working (may require manually refreshing doors). More details commented inside as there might just be a bug with doors and brokensprites in general.
-Fixed only some of the doors being damageable (consistency between doors) 


https://github.com/user-attachments/assets/7a55937a-045d-4447-8b48-06de1b921b1e

**Comment by ubertpendragon** [comment]
Alien door sprites are still not functional in unstable and you still have to lock connection pannels by giving them a tag manually 🙄 

Either your door is below 0.501 depth and the door frame occults the broken sprite, either it is above 0.499 and the background frame (including buttons) is rendered on top of characters

I fixed that in the doors I sent

![image](https://github.com/user-attachments/assets/bbccc569-5095-4a2f-a9f4-d66e54c59088)

**Comment by Regalis11** [comment]
These have been addressed now, closing
